### PR TITLE
Feature/tutorial donald

### PR DESCRIPTION
**Pull request by delippi**
Added a QC Ratio Check in the UFO. The user provides a min/max value to compare against a ratio. The Ratio Check rejects observations outside the user-provided bounds. A new ctest was also added.

All but 3 ctests passed during my first test. After fixing the conflicts, recompiling didn't work which was stated in the Slack that that is expected. Therefore, I was unable to rerun ctest a final time.

### CODE REVIEW

File: src/ufo/filters/QCmanager.cc [hunk 1]
```diff
@@ -110,6 +110,10 @@ void QCmanager::print(std::ostream & os) const {
     size_t itrack   = 0;
     size_t ibuddy   = 0;
     size_t ionedvar  = 0;
+<<<<<<< HEAD
+    size_t iratioref = 0;
+=======
+>>>>>>> origin/develop
```
Comment (aerorahul) [inline src/ufo/filters/QCmanager.cc:116]: please resolve conflicts

Comment (delippi) [reply src/ufo/filters/QCmanager.cc]: Thanks! That should be taken care of now.

File: src/ufo/filters/RatioCheck.cc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  std::vector<float> ref, val;
+  data_.get(ref_, ref);
+  data_.get(val_, val);
+  ASSERT(ref.size() == val.size());
+
+// Loop over all obs
+  for (size_t jobs = 0; jobs < nlocs; ++jobs) {
+    if (apply[jobs]) {
+      // check to see if one of the reference or value is missing or if ref (the denominator) is 0.
+      if (val[jobs] == missing || ref[jobs] == missing || ref[jobs] == 0) {
```
Comment (emilyhcliu) [inline src/ufo/filters/RatioCheck.cc:68]: @delippi The vector ref has a float type, so it is better to replace 0 with 0.0.

**Review by CoryMartin-NOAA — DISMISSED** [review]
other than 0 to 0.0, looks good

**Review by emilyhcliu — APPROVED** [review]

**Pull request closed without merging**